Scripts/CI/README_Metadata_StyleCheck/README_style_checker.py

```diff
@@ -0,0 +1,435 @@
+                f' - "{sets_diff}".')
+        # Check if all essential section headers present.
+        sets_diff = self.essential_headers - header_set
+        if sets_diff:
+            raise Exception(
+                f'Error header - Missing essential header(s) - "{sets_diff}".')
+        # Check if all sections are in correct order.
+        index = check_is_subsequence(self.readme_headers, self.possible_headers)
+        if index:
+            raise Exception(
+                f'Error header - Wrong order at - '
+                f'"{self.readme_headers[index-1]}".')
+
+    def check_format_title_section(self) -> None:
+        """
+        Check if
+        1. the head has at least 3 parts (title, description and image URLs).
+        2. the title string uses sentence case.
+        :return: None. Throws if exception occurs.
+        """
+        try:
+            title, _ = parse_head(self.readme_parts[0])
+            check_sentence_case(title)
+        except Exception as err:
+            raise Exception(f'Error title - {err}')
+
+    def check_format_apis(self) -> None:
+        """
+        Check if APIs
+        1. do not have backticks.
+        2. are sorted.
+        3. do not have duplicate entries.
+        :return: None. Throws if exception occurs.
+        """
+        try:
+            api_section_index = self.readme_parts.index('Relevant API') + 1
+            check_apis(self.readme_parts[api_section_index])
+        except Exception as err:
+            raise Exception(f'Error APIs - {err}')
+
+    def check_format_tags(self) -> None:
+        """
+        Check if tags
+        1. are in correct case.
+        2. are sorted.
+        3. do not have duplicate entries.
+        :return: None. Throws if exception occurs.
+        """
+        try:
+            tags_section_index = self.readme_parts.index('Tags') + 1
+            check_tags(self.readme_parts[tags_section_index])
+        except Exception as err:
+            raise Exception(f'Error tags - {err}')
+
+    def check_redundant_apis_in_tags(self) -> None:
+        """
+        Check if APIs and tags intersect.
+        :return: None. Throws if exception occurs.
+        """
+        try:
+            tags_section_index = self.readme_parts.index('Tags') + 1
+            api_section_index = self.readme_parts.index('Relevant API') + 1
+            api_set = check_apis(self.readme_parts[api_section_index])
+            tag_set = check_tags(self.readme_parts[tags_section_index])
+            if not api_set.isdisjoint(tag_set):
+                raise Exception(f'Error tags - API should not be in tags')
+        except Exception as err:
+            raise Exception(f'Error checking extra tags due to previous error')
+
+
+# region Main wrapper functions
+def run_check(path: str, count: int) -> int:
+    checker = ReadmeStyleChecker(path)
+    # 1. Populate from README.
+    try:
+        checker.populate_from_readme()
+    except Exception as err:
+        count += 1
```

Comment:
I can't tell if the actual number of exceptions is being used anywhere, so maybe it's possible to switch to `true`/`false`? 
I do recall there was an exit code 0/1 though, so maybe there's a technical reason for this as well.

Reply:
Yeah I think you're right. I think this is being output with the exception to show where the exception is e.g.
`1. map/display-map - Error title - README should contain title, description and image.` 
`1.` being the count. I guess because count is only 0/1 here where it compares `exception_count` to 0 it could be feasible to do true/false. But it maybe works with the exit code style? I'm hesitant to break this right now but could make a note in the CS issue if you think it'd be a good tidy up / improvement to explore?

Reply:
No worries, it's nothing major. Might be worth having a play with it, but if the errors are printed with a 'count' then that's ok too.